go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -464,6 +464,28 @@ func newSharedGatewayOpenFlowManager(nodeName, macAddress, gwBridge, gwIntf stri
 	}
 	nFlows++
 
+	if config.Gateway.DisableSNATMultipleGWs {
+		// table 1, traffic to pod subnet go directly to OVN
+		for _, clusterEntry := range config.Default.ClusterSubnets {
+			cidr := clusterEntry.CIDR
+			var ipPrefix string
+			if cidr.IP.To4() != nil {
+				ipPrefix = "ip"
+			} else {
+				ipPrefix = "ipv6"
+			}
+			mask, _ := cidr.Mask.Size()
+			_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
+				fmt.Sprintf("cookie=%s, priority=15, table=1, %s, %s_dst=%s/%d, actions=output:%s",
```

Comment:
I think this is intentional because mask is not the same as cidr.Mask. For this mask we use the full mask of an ipv4 or ipv6 address (/32 or /128) the cidr mask will be the subnet mask.

Reply:
I'm not sure I follow. The mask here is https://github.com/ovn-org/ovn-kubernetes/blob/2ab2e1de84c811496baee7e695e525bc59e513ee/go-controller/pkg/node/gateway_shared_intf.go#L477 so it's not the full mask but the subnet mask

Reply:
oh sorry you are right, I was thinking about the place I originally did this code and was using the entire mask. I'll change it and the other comment about the is ipv4 check.